### DISCUSSION TITLE
ci(): publish @next version on PR merge

### DISCUSSION THREAD

**Discussion by ShaMan123** [fabricjs/fabric.js#7256] [Ideas]
@asturur What do you think about adding a ci job that will build `dist` and publish to npm a `@next` tag after every PR merge?
I think it will be used and is a good devX addition.
I patch fabric manually when a major PR is merged to master (it is an error prone process because of other patches I maintain) so a `next` tag could be really useful for me and I guess for many others as well, plus it's good for testing and feedback.

**Comment by asturur** [comment]
that would be nice! i think is useful for many people.

**Reply by ShaMan123** [reply]
OK, how do we make this happen?
Git actions is a stranger to me.

**Comment by ShaMan123** [comment]
Docs:
https://github.com/actions/starter-workflows/tree/main/automation
https://docs.github.com/en/actions/creating-actions/about-actions

**Reply by asturur** [reply]
i think we can reuse our publish script and add an event for a commit on master

**Reply by asturur** [reply]
So we can duplicate this file, and call it `npmpublish-next`
https://github.com/fabricjs/fabric.js/blob/master/.github/workflows/npmpublish.yml

inside we change the `on:` condition with this one:

```yml
on:
  push:
    branches: [ master ]
```

we duplicate this file here
https://github.com/fabricjs/fabric.js/blob/master/publish.js

and we create one that is called publishnext.

Then we try it.

**Reply by ShaMan123** [reply]
Do you want to use the `beta` tag?
It has logic defined already. Or extends logic to accept any tag?
https://github.com/fabricjs/fabric.js/blob/master/publish.js#L13

**Reply by ShaMan123** [reply]
I see I can simply pass more args to the script file

**Reply by ShaMan123** [reply]
#7268 opened

**Comment by ShaMan123** [comment]
What do you think about a PR first comment action that builds a codesandbox fabric react app with the template I built #7178 ?

**Reply by ShaMan123** [reply]
I think codesandbox has an action out of the box for PRs
